### PR TITLE
Safer bid responses

### PR DESCRIPTION
**Pull request by janua**
## What does this change?

This makes the usage of `bidResponses` more safe by checking if the ad-slot ID exists before attempting to get the bids for it.

@guardian/commercial-dev

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js [hunk 1]
```diff
@@ -196,8 +196,14 @@ class PrebidService {
                                 const bidResponses = window.pbjs.getBidResponses()[
                                     advert.id
                                 ];
+                                const adSlotBidResponses =
+                                    bidResponses && advert.id in bidResponses
```
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: Could potentially use hasOwnProperty instead? 🤔 

`bidResponses  && bidResponses.hasOwnProperty(advert.id)`

**Review by katebee — APPROVED** [review]
👍 🚑 🏅 S A F E T Y

File: static/src/javascripts/projects/commercial/modules/prebid/prebid.js (hunk 1 above)
Comment (janua) [reply static/src/javascripts/projects/commercial/modules/prebid/prebid.js]: Yes, good shout! Not sure what's better here. Here we just care that it exists on the object or not! Using `hasOwnProperty` is assuming the object definitely does not inherit the key and we can't say for sure if that's true or not. (Although I imagine it's a fairly safe assumption to make! :smile: )

**Comment by philmcmahon** [comment]
This is over 3 months old now! What's gonna happen?

**Comment by SiAdcock** [comment]
Do we still need this @janua @katebee?

**Comment by SiAdcock** [comment]
Is there any update on this @katebee @janua?

**Comment by gtrufitt** [comment]
Can this 🚢 or 💀?

**Comment by janua** [comment]
This is no longer relevant.

**Comment by scottp719** [comment]
What an anticlimax

**Comment by gtrufitt** [comment]
Look who comes crawling back

**Comment by scottp719** [comment]
I'm still getting emails about this, so disappointed.

**Comment by janua** [comment]
😂

**Pull request closed without merging**